models/dataset.py

```diff
@@ -0,0 +1,184 @@
+from torch.utils.data import DataLoader, random_split
+import functools
+import h5py
+from PIL import Image
+CUDA = torch.cuda.is_available()
+
+if CUDA:
+    CRYO_TRAIN_VAL_DIR = os.getcwd()+"/Cryo/VAE_Cryo_V3/Data/"
+else:
+    CRYO_TRAIN_VAL_DIR = os.getcwd()+"\\Data\\"
+
+
+KWARGS = {'num_workers': 1, 'pin_memory': True} if CUDA else {}
+
+
+def open_dataset(path, new_size, D3):
+    """
+    Parameters
+    ----------
+    path : string path.
+    new_size : int.
+    D3 : boolean if 2d or 3d
+
+    Returns
+    -------
+    tensor of images in black and white.
+
+    """
+    if not os.path.exists(path):
+        raise OSError
+    if path[-3:] == '.h5':
+        data_dict = h5py.File(path, 'r')
+        all_datasets = data_dict['particles'][:]
+    else:
+        all_datasets = np.load(path)
+    dataset = np.asarray(all_datasets)
+    img_shape = dataset.shape
+    N = img_shape[0]
+    Dataset = []
+    if D3:
+        dataset = torch.Tensor(dataset)
+        dataset = normalization_linear(dataset)
+    else:
+        if len(img_shape) == 3:
+            for i in range(N):
+                Dataset.append(np.asarray(Image.fromarray(
+                    dataset[i]).resize([new_size, new_size])))
+        elif len(img_shape) == 4:
+            for i in range(N):
+                Dataset.append(np.asarray(Image.fromarray(
+                    dataset[i][0]).resize([new_size, new_size])))
+        dataset = torch.Tensor(Dataset)
+        dataset = normalization_linear(dataset)
+        if len(img_shape) != 4:
+            dataset = dataset.reshape(
+                (img_shape[0], 1, img_shape[1], img_shape[1]))
+    return dataset
+
+
+def normalization_linear(dataset):
+    """
+    Parameters
+    ----------
+    dataset : tensor of images.
+
+    Returns
+    -------
+    dataset : normalized tensor of images.
+
+    """
+    for i, data in enumerate(dataset):
+        min_data = torch.min(data)
+        max_data = torch.max(data)
+        if max_data != min_data:
+            raise ZeroDivisionError
+        dataset[i] = (data - min_data) / (max_data - min_data)
+    return dataset
+
+
+def organize_dataset(dataset, batch_size, frac_val):
+    """
+    Parameters
+    ----------
+    dataset : tensor of images.
+    batch_size : int batch_size.
+    frac_val : ratio between validation and training datasets.
+
+    Returns
+    -------
+    trainset : tensor of training images.
+    testset : tensor of test images.
+    trainloader : tensor ready to be used by the NN for training images.
+    testloader : tensor ready to be used by the NN for test images.
+
+    """
+    N = len(dataset)
+    n = int(N*frac_val)
+    trainset, testset = random_split(dataset, [N-n, n])
+    trainloader = DataLoader(
+        trainset, batch_size=batch_size, shuffle=True, **KWARGS)
+    testloader = DataLoader(
+        testset, batch_size=batch_size, shuffle=False, **KWARGS)
+    return trainset, testset, trainloader, testloader
+
+
+PATHS = {}
```

Comment:
Constants variables should go at the top of the file.

Reply:
Agreed. Also, I think Nina suggested storing this in a `.json` file, I think this is a good idea. 

@NicolasLegendre1 here is how you could do it:

# 1. Create a `dataset.json` file with the following content:
```json
{
  "PATHS": [
    {
      "simulated_2d": "cryo_sim_128x128.npy",
      "real_2d30": class2D_30_sort.h5",
      "real_2d39": "class2D_39_sort.h5",
      "real_2d93": "class2D_93_sort.h5",
      "refine_3D": "refine3D_180x180_sort.h5",
      "class_3d9090": "class3D_90x90_sort.h5",
      "simulated_3d": "concat_simulated.npy",
      "simulated_3d_noise": "cryo_sim_128x128.npy",
      "real_3d": "data.hdf5",
      "4points": "4points.npy",
      "4points1": "4points1.npy",
      "4points_3d": "3d_images.npy"
    }
  ],
  "SHAPES": [
    {
      "simulated_3d": (1, 320, 320),
      "simulated_2d": (1, 128, 128),
      "4points": (1, 128, 128),
      "4points1": (1, 64, 64),
      "4points_3d": (64, 64, 64)
    }
  ],
  etc...
}
```

# 2. add a function that loads it in `dataset.py`:
```python
import json

def load_parameters(path):
  with open(path) as json_file:
    parameters = json.load(json_file)
```

you can then get the values doing this, for example:
`filename = parameters['PATHS']['simulated_2d']`